src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -81,4 +80,16 @@ public ResponseEntity createDecoder(@RequestBody @Valid DecoderEntity decoderEnt
         }
         return ResponseEntity.ok("Decoder with serial number " + serialNumber + " Deleted");
     }
+
+    @PutMapping("/{serialNumber}")
+    public ResponseEntity<DecoderDTO> updateDecoder(@PathVariable String serialNumber,
+                                                    @RequestBody DecoderDTO decoderDTO){
+        Optional<DecoderEntity> decoder = decoderService.findDecoder(serialNumber);
+        if (decoder.isEmpty()) {
+            throw new ExceptionType.DeviceNotFoundException(serialNumber);
+        }
+        decoder.get().getInputs().clear();
```

Comment:
i wanted to make it explicit that:
9090 update to 9080 that updates the decoders port but doesnt update the port itself.

It could be removed, ill removed it